### PR TITLE
Connect register name to the API

### PR DESCRIPTION
**Pull request by solazio**
## Done
- _Connect register name to the API_
- _Fix `/charms` and `/bundles` page to show only the appropriate registered names_

## How to QA
- Run the project locally using the [dotrun](https://snapcraft.io/dotrun) snap with `$ dotrun` and view it in your web browser at: http://localhost:8045/register-name
- Register a bundle name and then a charm name. Go back to http://localhost:8045/charms and http://localhost:8045/bundles and see your names are displayed

## Issue / Card
Fixes https://github.com/canonical-web-and-design/snap-squad/issues/1917

## Screenshots

image

image

image

### CODE REVIEW

**Comment by tbille** [comment]
It seems that If you put an invalid name the page is 502. To reproduce:
- http://localhost:8045/register-name
- register name: `test-$`
- Returns 502

The API returns a 400. The error should be handled and passed to the frontend.

File: templates/publisher/register-name.html [hunk 1]
```diff
@@ -12,11 +12,56 @@ <h1 class="p-heading--2">
...
+      </div>
+    </div>
+  {% endif %}
+
+  {% if already_registered %}
+    <div class="row is-wide">
+      <div class="col-8">
+        <div class="p-notification--negative">
+          <p class="p-notification__response" role="status">
+            <span class="p-notification__status">Another publisher already registered Firefox.</span> You can email <a href="mailto:concierge@canonical.com">concierge@canonical.com</a> to file a dispute.
```
Comment (jkfran) [inline templates/publisher/register-name.html:34]: Always Firefox in the error message :(

File: templates/publisher/register-name.html [hunk 2]
```diff
@@ -12,11 +12,56 @@ <h1 class="p-heading--2">
...
-    <form class="p-form" action="">
-      <div>
+  </div>
+
+  {% if invalid_name %}
+    <div class="row is-wide">
+      <div class="col-8">
+        <div class="p-notification--negative">
+          <p class="p-notification__response" role="status">
+            <span class="p-notification__status">The name 'test-$' is not valid.</span> It should only have ASCII lowercase letters, numbers, and hyphens, and must have at least one letter.
```
Comment (jkfran) [inline templates/publisher/register-name.html:22]: 'test-$'?

File: templates/publisher/register-name.html (hunk 1 above)
Comment (solazio) [reply templates/publisher/register-name.html]: fixed. good catch

**Review by jkfran — APPROVED** [review]
LGTM

**Pull request merged**